homeassistant/components/droplet/sensor.py

```diff
@@ -0,0 +1,117 @@
+"""Support for Droplet."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pydroplet.droplet import Droplet
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import EntityCategory, UnitOfVolume, UnitOfVolumeFlowRate
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    KEY_CURRENT_FLOW_RATE,
+    KEY_SERVER_CONNECTIVITY,
+    KEY_SIGNAL_QUALITY,
+    KEY_VOLUME,
+)
+from .coordinator import DropletConfigEntry, DropletDataCoordinator
+
+
+@dataclass(kw_only=True, frozen=True)
+class DropletSensorEntityDescription(SensorEntityDescription):
+    """Describes Droplet sensor entity."""
+
+    value_fn: Callable[[Droplet], float | str | None]
+
+
+SENSORS: list[DropletSensorEntityDescription] = [
+    DropletSensorEntityDescription(
+        key=KEY_CURRENT_FLOW_RATE,
+        translation_key=KEY_CURRENT_FLOW_RATE,
+        device_class=SensorDeviceClass.VOLUME_FLOW_RATE,
+        native_unit_of_measurement=UnitOfVolumeFlowRate.LITERS_PER_MINUTE,
+        suggested_unit_of_measurement=UnitOfVolumeFlowRate.GALLONS_PER_MINUTE,
+        suggested_display_precision=2,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda device: device.get_flow_rate(),
+    ),
+    DropletSensorEntityDescription(
+        key=KEY_VOLUME,
+        translation_key=KEY_VOLUME,
```

Comment:
If you remove the translation key from an entity with a device class, it will use the device class translation, which helps with uniform naming of entities across intergations (if they mean the same) (and I would argue that in the case of Water volume it would mean the same as the device class)